argilla/src/argilla/settings/_io/_hub.py

```diff
@@ -0,0 +1,203 @@
+
+
+def _is_chat_feature(sub_features):
+    """Check if the sub_features correspond to an rg.ChatField."""
+    return (
+        "content" in sub_features
+        and "role" in sub_features
+        and sub_features["content"]["_type"] == "Value"
+        and sub_features["role"]["_type"] == "Value"
+        and sub_features["content"].get("dtype") == "string"
+        and sub_features["role"].get("dtype") == "string"
+    )
+
+
+def _define_settings_from_features(
+    features: Union[List[Dict], Dict[str, Any]], feature_mapping: Optional[Dict[str, str]]
+) -> "Settings":
+    """Define the argilla settings from the features of a dataset.
+
+    Parameters:
+        features (Dict[str, Dict[str, Any]): The features of the dataset.
+
+    Returns:
+        rg.Settings: The settings defined from the features.
+    """
+
+    from argilla import (
+        ImageField,
+        LabelQuestion,
+        TextQuestion,
+        Settings,
+        TextField,
+        TermsMetadataProperty,
+        IntegerMetadataProperty,
+        FloatMetadataProperty,
+    )
+
+    fields = []
+    questions = []
+    metadata = []
+    mapping = defaultdict(list)
+    feature_mapping = feature_mapping or {}
+    mapped_questions = [key for key, value in feature_mapping.items() if value == "question"]
+    mapped_fields = [key for key, value in feature_mapping.items() if value == "field"]
+    mapped_metadata = [key for key, value in feature_mapping.items() if value == "metadata"]
+
+    for name, feature in features.items():
+        feature_type = _map_feature_type(feature)
+
+        if feature_type == FeatureType.CHAT:
+            # TODO: Implement chat support to create `rg.ChatField`
+            # fields.append(rg.ChatField(name=f"{name}_field"))
+            # mapping[name] = f"{name}_field"
+            pass
+
+        elif feature_type == FeatureType.TEXT:
+            if name not in mapped_questions:
+                fields.append(TextField(name=name))
+                mapping[name].append(name)
+            if name not in mapped_fields:
+                questions.append(TextQuestion(name=f"{name}_question"))
+                mapping[name].append(f"{name}_question")
+
+        elif feature_type == FeatureType.IMAGE:
+            fields.append(ImageField(name=name))
+
+        elif feature_type == FeatureType.LABEL:
+            names = feature.get("names")
+            if names is None:
+                warnings.warn(f"Feature '{name}' has no labels. Skipping.")
+                continue
+            if name not in mapped_metadata:
+                questions.append(
+                    LabelQuestion(
+                        name=name,
+                        labels=names,
+                    )
+                )
+                mapping[name].append(name)
+            if name not in mapped_questions:
+                metadata.append(TermsMetadataProperty(name=f"{name}_metadata", options=names))
```

Comment:
Wouldn't it be easier to potentially allow for all options? Similar to the approach with Integers and Floats?

Reply:
So to leave the `options` param empty? I always thought defining options improved search filters UX, but I might be wrong.

Reply:
I am not entirely sure @frascuchon

Reply:
The backend supplies unique values if the options are not defined. So I've gone with this.